src/BinSkim.Rules/PERules/BA2027.CompileUsingCurrentDialects.cs

```diff
@@ -0,0 +1,193 @@
+                }
+                else
+                {
+                    continue;
+                }
+
+                if (!details.HasDebugInfo)
+                {
+                    continue;
+                }
+
+                string cVersionNumberString = string.Empty;
+                details.GetOptionValue(cVersion, OrderOfPrecedence.FirstWins, ref cVersionNumberString, cVersionExecluded);
+                AnalyzeResultType resultType = AnalyzeResultType.NotCurrent;
+
+                if (string.IsNullOrWhiteSpace(cVersionNumberString))
+                {
+                    resultType = AnalyzeResultType.NotSet;
+                }
+                else
+                {
+                    int cVersionNumber;
+                    if (cVersionNumberString.Equals("latest", StringComparison.OrdinalIgnoreCase))
+                    {
+                        resultType = AnalyzeResultType.Latest;
+                    }
+                    else if (int.TryParse(cVersionNumberString, out cVersionNumber))
+                    {
+                        resultType = cVersionNumber >= 17 ? AnalyzeResultType.Pass : AnalyzeResultType.NotCurrent;
+                    }
+                    else
+                    {
+                        resultType = AnalyzeResultType.Unsupported;
+                    }
+                }
+
+                if (resultType != AnalyzeResultType.Pass && resultType != AnalyzeResultType.Latest)
+                {
+                    if (!notCompiledUsingCurrentDialectsModules.Any(l => l.Details.Library == details.Library))
+                    {
+                        notCompiledUsingCurrentDialectsModules.Add(new AnalyzeResult() { Details = details, Version = cVersionNumberString, Type = resultType });
+                    }
+                }
+            }
+
+            if (notCompiledUsingCurrentDialectsModules.Count > 0)
+            {
+                string line;
+                var sb = new StringBuilder();
+
+                foreach (AnalyzeResult result in notCompiledUsingCurrentDialectsModules)
+                {
+                    // Library: {0}, Language: {1}, Version: {2}
+                    line = string.Format(RuleResources.BA2027_Warning_Item,
+                        result.Details.Library,
+                        result.Details.Language,
+                        result.Type == AnalyzeResultType.NotSet ? "NotSet" : result.Version);
+                    sb.AppendLine(line);
+                }
+
+                // '{0}' is a Windows PE that wasn't compiled with current dialects.
+                // Compile using current dialects enables current standard-specific features and behavior.
+                // To resolve this problem, pass version 17 or later on the cl.exe command-line,
+                // e.g. '/std:c++17' for C++ and '/std:c17' for C 
+                // or set the corresponding 'Language Standard' property in the 'C/C++ -> General' Configuration property page.
```

Comment:
> or set  [](http://example.com/codeflow?start=18&length=8)

Check the other rules to get some examples and the horizontal length that we normally use.
The text does not look right and there are lines that are too long

Reply:
I looked the BA2021 how long it is and using the same as that one, align.